src/js/binaryen.js-post.js

```diff
@@ -30,442 +30,436 @@ function i8sToStack(i8s) {
+  Module['MulVecF64x2'] = Module['_BinaryenMulVecF64x2']();
+  Module['DivVecF64x2'] = Module['_BinaryenDivVecF64x2']();
+  Module['MinVecF64x2'] = Module['_BinaryenMinVecF64x2']();
+  Module['MaxVecF64x2'] = Module['_BinaryenMaxVecF64x2']();
+  Module['TruncSatSVecF32x4ToVecI32x4'] = Module['_BinaryenTruncSatSVecF32x4ToVecI32x4']();
+  Module['TruncSatUVecF32x4ToVecI32x4'] = Module['_BinaryenTruncSatUVecF32x4ToVecI32x4']();
+  Module['TruncSatSVecF64x2ToVecI64x2'] = Module['_BinaryenTruncSatSVecF64x2ToVecI64x2']();
+  Module['TruncSatUVecF64x2ToVecI64x2'] = Module['_BinaryenTruncSatUVecF64x2ToVecI64x2']();
+  Module['ConvertSVecI32x4ToVecF32x4'] = Module['_BinaryenConvertSVecI32x4ToVecF32x4']();
+  Module['ConvertUVecI32x4ToVecF32x4'] = Module['_BinaryenConvertUVecI32x4ToVecF32x4']();
+  Module['ConvertSVecI64x2ToVecF64x2'] = Module['_BinaryenConvertSVecI64x2ToVecF64x2']();
+  Module['ConvertUVecI64x2ToVecF64x2'] = Module['_BinaryenConvertUVecI64x2ToVecF64x2']();
+  Module['LoadSplatVec8x16'] = Module['_BinaryenLoadSplatVec8x16']();
+  Module['LoadSplatVec16x8'] = Module['_BinaryenLoadSplatVec16x8']();
+  Module['LoadSplatVec32x4'] = Module['_BinaryenLoadSplatVec32x4']();
+  Module['LoadSplatVec64x2'] = Module['_BinaryenLoadSplatVec64x2']();
+  Module['LoadExtSVec8x8ToVecI16x8'] = Module['_BinaryenLoadExtSVec8x8ToVecI16x8']();
+  Module['LoadExtUVec8x8ToVecI16x8'] = Module['_BinaryenLoadExtUVec8x8ToVecI16x8']();
+  Module['LoadExtSVec16x4ToVecI32x4'] = Module['_BinaryenLoadExtSVec16x4ToVecI32x4']();
+  Module['LoadExtUVec16x4ToVecI32x4'] = Module['_BinaryenLoadExtUVec16x4ToVecI32x4']();
+  Module['LoadExtSVec32x2ToVecI64x2'] = Module['_BinaryenLoadExtSVec32x2ToVecI64x2']();
+  Module['LoadExtUVec32x2ToVecI64x2'] = Module['_BinaryenLoadExtUVec32x2ToVecI64x2']();
+  Module['NarrowSVecI16x8ToVecI8x16'] = Module['_BinaryenNarrowSVecI16x8ToVecI8x16']();
+  Module['NarrowUVecI16x8ToVecI8x16'] = Module['_BinaryenNarrowUVecI16x8ToVecI8x16']();
+  Module['NarrowSVecI32x4ToVecI16x8'] = Module['_BinaryenNarrowSVecI32x4ToVecI16x8']();
+  Module['NarrowUVecI32x4ToVecI16x8'] = Module['_BinaryenNarrowUVecI32x4ToVecI16x8']();
+  Module['WidenLowSVecI8x16ToVecI16x8'] = Module['_BinaryenWidenLowSVecI8x16ToVecI16x8']();
+  Module['WidenHighSVecI8x16ToVecI16x8'] = Module['_BinaryenWidenHighSVecI8x16ToVecI16x8']();
+  Module['WidenLowUVecI8x16ToVecI16x8'] = Module['_BinaryenWidenLowUVecI8x16ToVecI16x8']();
+  Module['WidenHighUVecI8x16ToVecI16x8'] = Module['_BinaryenWidenHighUVecI8x16ToVecI16x8']();
+  Module['WidenLowSVecI16x8ToVecI32x4'] = Module['_BinaryenWidenLowSVecI16x8ToVecI32x4']();
+  Module['WidenHighSVecI16x8ToVecI32x4'] = Module['_BinaryenWidenHighSVecI16x8ToVecI32x4']();
+  Module['WidenLowUVecI16x8ToVecI32x4'] = Module['_BinaryenWidenLowUVecI16x8ToVecI32x4']();
+  Module['WidenHighUVecI16x8ToVecI32x4'] = Module['_BinaryenWidenHighUVecI16x8ToVecI32x4']();
+  Module['SwizzleVec8x16'] = Module['_BinaryenSwizzleVec8x16']();
```

Comment:
Is this all maintained by hand?  It seems pretty fragile?  Can this not be done by looking at all the keys in Module that start with `_Binaryen`?

Reply:
Yeah, that's all maintained by hand. I believe one reason why it's hard to automate is that not every exported function of the form `_Binaryen...` can be called like that, for example some of these return the respective constants while others like `_BinaryenModuleCreate` create a module. But might be possible to simplify by keeping a list of the good ones, doing something like

```js
["WidenHighSVecI16x8ToVecI32x4", ...].forEach(name => Module[name] = Module['_Binaryen' + name]());
```
```